### DISCUSSION TITLE
Why does PnP Core SDK try to parse string columns as DateTime? How can I prevent it?

### DISCUSSION THREAD

**Discussion by jesuslpm** [pnp/pnpcore#1431] [Q&A]
I have the following list:

<img width="619" alt="MyList" src="https://github.com/pnp/pnpcore/assets/2075727/6b91d305-0163-4e2c-b641-945056368999">

Column1, Column2 and Column3 are text columns. However this code:

```
public void ShowMyList()
{
    var list = this.context.Web.Lists.GetByTitle("MyList");
    foreach (var item in list.Items)
    {
        Console.WriteLine($"Id: {item.Id}");
        Console.WriteLine($"Title: {item.Title}");
        var col1 = item.Values["Column1"];
        Console.WriteLine($"Column1 Value: {Convert.ToString(col1, CultureInfo.InvariantCulture)}");
        Console.WriteLine($"Column1 Type: {col1.GetType().FullName}");

        var col2 = item.Values["Column2"];
        Console.WriteLine($"Column2 Value: {Convert.ToString(col2, CultureInfo.InvariantCulture)}");
        Console.WriteLine($"Column2 Type: {col2.GetType().FullName}");


        var col3 = item.Values["Column3"];
        Console.WriteLine($"Column3 Value: {Convert.ToString(col3, CultureInfo.InvariantCulture)}");
        Console.WriteLine($"Column3 Type: {col3.GetType().FullName}");
    }
}
```
Outputs this:

> Id: 1
> Title: This is the title
> 
> Column1 Value: 03/24/2024 00:00:00
> Column1 Type: System.DateTime
> 
> Column2 Value: 03/24/2024 07:50:00
> Column2 Type: System.DateTime
> 
> Column3 Value: 2024-03-24-699
> Column3 Type: System.String

I don't understand why this library tries to parse those strings as DateTime, they are text columns. How can I get the text stored on those columns? How can I prevent PnP Core from parsing them?

**Comment by jansenbe** [comment]
@jesuslpm : currently this is "by design", see https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/JsonMappingHelper.cs#L1049-L1058. Think we did this as in early iterations there were situations where we did not have the field information present. I'll review things and check if this can be changed, if so it will be an opt-in change as such a change could break the code from others. 

For now it's best you convert back to string

**Reply by jesuslpm** [reply]
> there were situations where we did not have the field information present

In this case you have the column names and types

>  I'll review things and check if this can be changed

Thank you.

> For now it's best you convert back to string

I think this is not always possible. How do you get the original string from a DateTime reliably?
